### PR TITLE
Feat/auth based on group membership and admin-page

### PR DESCRIPTION
**Pull request by olros**
# Feat/Page-route group-auth
## Auth based on group membership and admin-page
Link to Trello task: https://trello.com/c/jmbO06fL/158-admin-route-og-admin-side-fix

Comments/issues:

## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] PR name is descriptive
- [x] The PR includes the link to the relevant Trello task
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)

- [ ] `yarn build` was run locally witout errors
- [ ] `yarn test` has passed locally and any fixes were made for failures

## Pull request type

Please check the type of change your PR introduces:
- [x] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

### CODE REVIEW

File: src/api/services/UserService.js [hunk 1]
```diff
@@ -25,6 +25,27 @@ class UserService {
...
+                        case "Promo": isPromo = true; break;
+                        case "NoK": isNok = true; break;
+                        case "DevKom": isDevkom = true; break;
+                        default: break;
+                    }
+                });
+                return {"isHS":isHS,"isPromo":isPromo,"isNok":isNok,"isDevkom":isDevkom};
+            });
+        } else {
+            return{"isHS":isHS,"isPromo":isPromo,"isNok":isNok,"isDevkom":isDevkom};
```
Comment (Zenjjim) [inline src/api/services/UserService.js]: Kan du ikke bare legge return utenfor if elsen siden det returner det samme

File: src/containers/Admin/index.js [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+        super(props);
+        this.state = {
+            isHS: false,
+            isPromo: false,
+            isNok: false,
+            isDevkom: false,
+        }
+    }
+
+    loadIsGroupMember = () => {
```
Comment (Zenjjim) [inline src/containers/Admin/index.js]: bruh... you know

File: src/index.js [hunk 3]
```diff
@@ -32,22 +33,52 @@ import Laws from './containers/Laws';
...
+                        case "promo": if(groups.isPromo){ this.setState({allowAccess: true}); }; break;
+                        case "nok": if(groups.isNok){ this.setState({allowAccess: true}); }; break;
+                        case "devkom": if(groups.isDevkom){ this.setState({allowAccess: true}); }; break;
+                        default: break;
+                    }
+                });
+                this.setState({isLoading: false});
+            }).catch(() => {
+                this.setState({isLoading: false});
+            })
```
Comment (Zenjjim) [inline src/index.js]: samme ^

File: src/api/services/UserService.js (hunk 1 above)
Comment (Andorr) [reply src/api/services/UserService.js]: Den returnerer inne i ".then" metoden, som er noe annet. Men han kan for eksempel la objektet være en egen variabel, så endre den i ".then"-metoden, og deretter returnere bare en gang

File: src/index.js (hunk 3 above)
Comment (Andorr) [reply src/index.js]: Blir bruk av ".finally(() => this.setState({...}))" her ^^ 👍

**Pull request merged**